### PR TITLE
fix(@jest/types): align jest v28 dependencies (requires typescript bump to v4.5)

### PR DESCRIPTION
**Pull request by viceice**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by viceice** [comment]
This is a folowup from #58784. If you upgrade all jest packages to v28, typescript will complain because of importing wrong jest types

**Comment by viceice** [comment]
@SimenB WDYT?

**Comment by viceice** [comment]
```
[11:02:17 AM] Starting compilation in watch mode...

test/util.ts:141:32 - error TS2345: Argument of type 'Plugin' is not assignable to parameter of type 'Plugin_2'.
  Type 'NewPlugin' is not assignable to type 'Plugin_2'.
    Type 'import("/home/kriese/src/renovate/node_modules/@types/jest/node_modules/pretty-format/build/types").NewPlugin' is not assignable to type 'import("/home/kriese/src/renovate/node_modules/pretty-format/build/index").NewPlugin'.
      Types of property 'serialize' are incompatible.
        Type '(val: any, config: import("/home/kriese/src/renovate/node_modules/@types/jest/node_modules/pretty-format/build/types").Config, indentation: string, depth: number, refs: import("/home/kriese/src/renovate/node_modules/@types/jest/node_modules/pretty-format/build/types").Refs, printer: import("/home/kriese/src/renovate...' is not assignable to type '(val: any, config: import("/home/kriese/src/renovate/node_modules/pretty-format/build/index").Config, indentation: string, depth: number, refs: import("/home/kriese/src/renovate/node_modules/pretty-format/build/index").Refs, printer: import("/home/kriese/src/renovate/node_modules/pretty-format/build/index").Printer)...'.
          Types of parameters 'config' and 'config' are incompatible.
            Type 'import("/home/kriese/src/renovate/node_modules/pretty-format/build/index").Config' is not assignable to type 'import("/home/kriese/src/renovate/node_modules/@types/jest/node_modules/pretty-format/build/types").Config'.
              Types of property 'plugins' are incompatible.
                Type 'import("/home/kriese/src/renovate/node_modules/pretty-format/build/index").Plugins' is not assignable to type 'import("/home/kriese/src/renovate/node_modules/@types/jest/node_modules/pretty-format/build/types").Plugins'.
                  Type 'Plugin_2' is not assignable to type 'Plugin'.
                    Type 'NewPlugin' is not assignable to type 'Plugin'.
                      Type 'import("/home/kriese/src/renovate/node_modules/pretty-format/build/index").NewPlugin' is not assignable to type 'import("/home/kriese/src/renovate/node_modules/@types/jest/node_modules/pretty-format/build/types").NewPlugin'.
                        Types of property 'serialize' are incompatible.
                          Type '(val: any, config: import("/home/kriese/src/renovate/node_modules/pretty-format/build/index").Config, indentation: string, depth: number, refs: import("/home/kriese/src/renovate/node_modules/pretty-format/build/index").Refs, printer: import("/home/kriese/src/renovate/node_modules/pretty-format/build/index").Printer)...' is not assignable to type '(val: any, config: import("/home/kriese/src/renovate/node_modules/@types/jest/node_modules/pretty-format/build/types").Config, indentation: string, depth: number, refs: import("/home/kriese/src/renovate/node_modules/@types/jest/node_modules/pretty-format/build/types").Refs, printer: import("/home/kriese/src/renovate...'.
                            Types of parameters 'config' and 'config' are incompatible.
                              Property 'maxWidth' is missing in type 'import("/home/kriese/src/renovate/node_modules/@types/jest/node_modules/pretty-format/build/types").Config' but required in type 'import("/home/kriese/src/renovate/node_modules/pretty-format/build/index").Config'.

141   expect.addSnapshotSerializer(replacingSerializer(from, to));
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/pretty-format/build/index.d.ts:44:3
    44   maxWidth: number;
         ~~~~~~~~
    'maxWidth' is declared here.

test/util.ts:157:32 - error TS2345: Argument of type 'Plugin' is not assignable to parameter of type 'Plugin_2'.

157   expect.addSnapshotSerializer(bufferSerializer);
                                   ~~~~~~~~~~~~~~~~

[11:02:34 AM] Found 2 errors. Watching for file changes.
```

- https://github.com/renovatebot/renovate/pull/16103

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## jest-when ([<kbd>unpkg</kbd>](https://unpkg.com/browse/jest-when@latest/))
was missing the following properties:
1. WhenMock
1. WhenMock
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 71a25e64e355683eac27fa36e7515d5a945dab95
</p>

File: types/jest/index.d.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@
 //                 Alexandre Germain <https://github.com/gerkindev>
 //                 Adam Jones <https://github.com/domdomegg>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// Minimum TypeScript Version: 3.8
+// Minimum TypeScript Version: 4.5
```
Comment (SimenB) [inline types/jest/index.d.ts:31]: 4.3, not 4.5 🙂

File: types/expect-puppeteer/index.d.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@
 // Definitions by: Tanguy Krotoff <https://github.com/tkrotoff>
 //                 Jason Mong <https://github.com/jfm710>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/expect-puppeteer/index.d.ts:6]:
```suggestion
// TypeScript Version: 4.3
```

File: types/heft-jest/index.d.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@
 // Definitions by: Pete Gonzalez <https://github.com/octogonz>
 //                 Ian Clanton-Thuon <https://github.com/iclanton>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/heft-jest/index.d.ts:6]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest-axe/index.d.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/nickcolley/jest-axe
 // Definitions by: erbridge <https://github.com/erbridge>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/jest-axe/index.d.ts:5]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest-expect-message/index.d.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@
 // Definitions by: Mike Davydov <https://github.com/mike-d-davydov>
 //                 Michael Bashurov <https://github.com/saitonakamura>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/jest-expect-message/index.d.ts:6]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest-image-snapshot/index.d.ts [hunk 6]
```diff
@@ -4,7 +4,7 @@
 //                 erbridge <https://github.com/erbridge>
 //                 Piotr Błażejewicz <https://github.com/peterblazejewicz>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/jest-image-snapshot/index.d.ts:7]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest-in-case/index.d.ts [hunk 7]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/thinkmill/jest-in-case#readme
 // Definitions by: Geovani de Souza <https://github.com/geovanisouza92>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/jest-in-case/index.d.ts:5]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest-json-schema/index.d.ts [hunk 8]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/americanexpress/jest-json-schema#readme
 // Definitions by: Matt Scheurich <https://github.com/lvl99>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/jest-json-schema/index.d.ts:5]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest-matcher-one-of/index.d.ts [hunk 9]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/d4nyll/jest-matcher-one-of#readme
 // Definitions by: Joe Mitchard <https://github.com/joemitchard>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/jest-matcher-one-of/index.d.ts:5]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest-plugin-context/index.d.ts [hunk 10]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/negativetwelve/jest-plugins/tree/master/packages/jest-plugin-context, https://github.com/negativetwelve/jest-plugins
 // Definitions by: Jonas Heinrich <https://github.com/jonasheinrich>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/jest-plugin-context/index.d.ts:5]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest-sinon/index.d.ts [hunk 11]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/djkf/jest-sinon/#readme
 // Definitions by: Martin Seidel <https://github.com/seidelmartin>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/jest-sinon/index.d.ts:5]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest-specific-snapshot/index.d.ts [hunk 12]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/igor-dv/jest-specific-snapshot#readme
 // Definitions by: Janeene Beeforth <https://github.com/dawnmist>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/jest-specific-snapshot/index.d.ts:5]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest-when/index.d.ts [hunk 13]
```diff
@@ -5,7 +5,7 @@
 //                 Gregor Stamać <https://github.com/gstamac>
 //                 Nicholas Hehr <https://github.com/hipsterbrown>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/jest-when/index.d.ts:8]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest/index.d.ts (hunk 1 above)
Comment (viceice) [reply types/jest/index.d.ts:31]:
```suggestion
// TypeScript Version: 4.3
```

File: types/mapbox__aws-sdk-jest/index.d.ts [hunk 14]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/mapbox/aws-sdk-jest#readme
 // Definitions by: stevensnoeijen <https://github.com/stevensnoeijen>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// Minimum TypeScript Version: 3.8
+// Minimum TypeScript Version: 4.5
```
Comment (viceice) [inline types/mapbox__aws-sdk-jest/index.d.ts:5]:
```suggestion
// TypeScript Version: 4.3
```

File: types/testing-library__jest-dom/index.d.ts [hunk 15]
```diff
@@ -5,7 +5,7 @@
 //                 Seth Macpherson <https://github.com/smacpherson64>
 //                 Andrew Leedham <https://github.com/AndrewLeedham>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.5
```
Comment (viceice) [inline types/testing-library__jest-dom/index.d.ts:8]:
```suggestion
// TypeScript Version: 4.3
```

File: types/jest/index.d.ts (hunk 1 above)
Comment (viceice) [reply types/jest/index.d.ts]: I simply used what `@sinclair/typebox` was referencing to 😉

File: types/jest-json-schema/index.d.ts [hunk 16]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/americanexpress/jest-json-schema#readme
 // Definitions by: Matt Scheurich <https://github.com/lvl99>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.3
```
Comment (viceice) [inline types/jest-json-schema/index.d.ts:5]:
```suggestion
// Minimum TypeScript Version: 4.3
```

File: types/jest-matcher-one-of/index.d.ts [hunk 17]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/d4nyll/jest-matcher-one-of#readme
 // Definitions by: Joe Mitchard <https://github.com/joemitchard>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.3
```
Comment (viceice) [inline types/jest-matcher-one-of/index.d.ts:5]:
```suggestion
// Minimum TypeScript Version: 4.3
```

File: types/jest-plugin-context/index.d.ts [hunk 18]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/negativetwelve/jest-plugins/tree/master/packages/jest-plugin-context, https://github.com/negativetwelve/jest-plugins
 // Definitions by: Jonas Heinrich <https://github.com/jonasheinrich>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.3
```
Comment (viceice) [inline types/jest-plugin-context/index.d.ts:5]:
```suggestion
// Minimum TypeScript Version: 4.3
```

File: types/jest-sinon/index.d.ts [hunk 19]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/djkf/jest-sinon/#readme
 // Definitions by: Martin Seidel <https://github.com/seidelmartin>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.3
```
Comment (viceice) [inline types/jest-sinon/index.d.ts:5]:
```suggestion
// Minimum TypeScript Version: 4.3
```

File: types/jest-specific-snapshot/index.d.ts [hunk 20]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/igor-dv/jest-specific-snapshot#readme
 // Definitions by: Janeene Beeforth <https://github.com/dawnmist>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.3
```
Comment (viceice) [inline types/jest-specific-snapshot/index.d.ts:5]:
```suggestion
// Minimum TypeScript Version: 4.3
```

File: types/jest-when/index.d.ts [hunk 21]
```diff
@@ -5,7 +5,7 @@
 //                 Gregor Stamać <https://github.com/gstamac>
 //                 Nicholas Hehr <https://github.com/hipsterbrown>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.8
+// TypeScript Version: 4.3
```
Comment (viceice) [inline types/jest-when/index.d.ts:8]:
```suggestion
// Minimum TypeScript Version: 4.3
```

File: types/jest/index.d.ts [hunk 22]
```diff
@@ -28,7 +28,7 @@
 //                 Alexandre Germain <https://github.com/gerkindev>
 //                 Adam Jones <https://github.com/domdomegg>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// Minimum TypeScript Version: 3.8
+// TypeScript Version: 4.3
```
Comment (viceice) [inline types/jest/index.d.ts:31]:
```suggestion
// Minimum TypeScript Version: 4.3
```

File: types/mapbox__aws-sdk-jest/index.d.ts [hunk 23]
```diff
@@ -2,7 +2,7 @@
 // Project: https://github.com/mapbox/aws-sdk-jest#readme
 // Definitions by: stevensnoeijen <https://github.com/stevensnoeijen>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// Minimum TypeScript Version: 3.8
+// TypeScript Version: 4.3
```
Comment (viceice) [inline types/mapbox__aws-sdk-jest/index.d.ts:5]:
```suggestion
// Minimum TypeScript Version: 4.3
```

**Review by SimenB — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by viceice** [comment]
Ready to merge

**Pull request merged**